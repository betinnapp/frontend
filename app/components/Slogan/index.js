/**
 *
 * Slogan
 *
 */

import React from 'react'
import { FormattedMessage } from 'react-intl'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Icon from 'images/icon.svg'
import messages from 'containers/App/messages'

const Wrapper = styled.div`
  display: inline-block;
  font-size: ${props => (props.big ? '28px' : '16px')};
  font-weight: bold;
  text-transform: uppercase;

  img {
    width: ${props => (props.big ? '96px' : '46px')};
    margin-right: ${props => (props.big ? '20px' : '10px')};
  }
`

function Slogan({ big }) {
  return (
    <Wrapper big={big}>
      <img src={Icon} alt="Betinnapp icon" />
      <FormattedMessage {...messages.betinnapp} />
    </Wrapper>
  )
}

Slogan.propTypes = {
  big: PropTypes.bool,
}

Slogan.defaultProps = {
  big: false,
}

export default Slogan
