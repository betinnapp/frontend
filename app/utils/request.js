import axios from 'axios'

export default function request(url, options) {
  const newOptions = {
    url,
    ...options,
  }

  if (!options.headers) {
    newOptions.headers = {
      'Content-type': 'application/json',
    }
  }

  return axios(newOptions).then(response => response.data)
}
