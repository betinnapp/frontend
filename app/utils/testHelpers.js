import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'

import ptTranslations from 'translations/pt.json'

export function renderWithIntl(component) {
  return render(
    <IntlProvider locale="pt" messages={ptTranslations}>
      {component}
    </IntlProvider>,
  )
}
