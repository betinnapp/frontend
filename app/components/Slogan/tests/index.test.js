/**
 *
 * Tests for Slogan
 *
 */

import React from 'react'
import { renderWithIntl } from 'utils/testHelpers'

import Slogan from '../index'

describe('<Slogan />', () => {
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderWithIntl(<Slogan />)
    expect(firstChild).toMatchSnapshot()
  })
})
