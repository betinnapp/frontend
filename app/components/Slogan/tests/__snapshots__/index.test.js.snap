// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slogan /> Should render and match the snapshot 1`] = `
<div
  class="Slogan__Wrapper-v1mhy0-0 dhXgpb"
>
  <img
    alt="Betinnapp icon"
    src="IMAGE_MOCK"
  />
  <span>
    Betinnapp
  </span>
</div>
`;
